portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,9 +35,22 @@
 public class DataServlet extends HttpServlet {
 
   private List<String> comments = new ArrayList<>();
+  // Declares instance of Datastore Service
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Prepares results and sorts them in descending order.
+    Query query = new Query("entry").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    comments.clear();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("com");
+      String firstN = (String) entity.getProperty("firstN");
+      String lastN = (String) entity.getProperty("lastN");
```

Comment:
What happens if any of these are empty?

Reply:
In this PR, the entry in the comment display would leave the corresponding space blank. But this is fixed in a following PR!